### PR TITLE
[SG-33258]: Wildcard `<Input />` migration

### PR DESCRIPTION
**Pull request by gitstart-sourcegraph**
## Descripton
In order to preserve layout needed for some css rules such as `form-group > .form-control`, to be appropriately applied on the wildcard `Input` component, the modification was made, to separate the support for `LoaderInput` from the default Input component

## Ref
[SG Parent Issue](https://github.com/sourcegraph/sourcegraph/issues/33258)
[SG Issue](https://github.com/sourcegraph/sourcegraph/issues/35775)
[Gitstart Ticket](https://app.gitstart.com/clients/sourcegraph/tickets/SG-35775)

## Test Plan
View components on storybook to ensure expected behavior

## App preview:

- [Web](https://sg-web-contractors-sg-33258.onrender.com)
- [Storybook](https://5f0f381c0e50750022dc6bf7-tlmmjqyfxx.chromatic.com)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

**Comment by gitstart-sourcegraph** [comment]
Hi @valerybugakov 
Help take a look at changes here, for updating the layout of `Input` component as discussed.
WDYT about usage of `Input` and `FormInput`, with regards to how intuitive their use cases would be to component users.

**Review by valerybugakov — CHANGES_REQUESTED** [review]
See the comment on the `FormInput` implementation.

File: client/wildcard/src/components/Form/Input/FormInput.tsx [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+import { useAutoFocus } from '../../../hooks/useAutoFocus'
+import { ForwardReferenceComponent } from '../../../types'
+import { Label } from '../../Typography/Label'
+
+import { InputProps } from './Input'
+
+import styles from './Input.module.scss'
+
+export enum FormInputStatus {
```
Comment (valerybugakov) [inline client/wildcard/src/components/Form/Input/FormInput.tsx:16]: The `FormInputStatus` should extend the `InputStatus` with the `loading` property.

**Comment by gitstart-sourcegraph** [comment]
Paused since this implementation might not be very intuitive for component users.
Would be re-visited after migration efforts in #35579

**Pull request closed without merging**